lib/include/tkge/Assets/TextAsset.hpp

```diff
@@ -4,11 +4,20 @@
 
 namespace tkge::Assets
 {
-	class TextAsset final : public ICopyableAsset<TextAsset>
+	class TextAsset final : public IAsset
 	{
 	  public:
-		explicit TextAsset(std::string filename) : ICopyableAsset(std::move(filename)) {}
-		[[nodiscard]] std::string ReadAllText() const;
-		[[nodiscard]] std::string ReadAt(std::size_t position, std::size_t size) const;
+		explicit TextAsset() = default;
+		bool Load(ReadonlyByteStream byteStream) override
+		{
+			const auto vData = byteStream.AsSpan<char>(0, byteStream.GetStreamSize());
+			this->_text = std::string{vData.begin(), vData.end()};
+			return true;
+		}
```

Comment:
Better to keep multi-line impls in cpp files unless there's good reason to pollute headers.